src/bundle/Resources/config/services.yaml

```
@@ -1,25 +1,49 @@
 imports:
-  - { resource: forms.yaml }
-  - { resource: twig.yaml }
-  - { resource: sorting_definitions.yaml }
-  - { resource: views.yaml }
+    - { resource: forms.yaml }
+    - { resource: twig.yaml }
+    - { resource: sorting_definitions.yaml }
+    - { resource: views.yaml }
 
 services:
-  _defaults:
-      autoconfigure: true
-      autowire: true
-      public: false
-
-  Ibexa\Bundle\Search\Controller\SearchController:
-      tags:
-          - controller.service_arguments
-
-  Ibexa\Search\Mapper\PagerSearchContentToDataMapper:
-      arguments:
-          $contentTypeService: '@ibexa.api.service.content_type'
-          $userService: '@ibexa.api.service.user'
-          $userLanguagePreferenceProvider: '@Ibexa\Core\MVC\Symfony\Locale\UserLanguagePreferenceProvider'
-          $translationHelper: '@Ibexa\Core\Helper\TranslationHelper'
-          $languageService: '@ibexa.api.service.language'
-
-  Ibexa\Search\QueryType\SearchQueryType: ~
+    _defaults:
+        autoconfigure: true
+        autowire: true
+        public: false
+
+    Ibexa\Bundle\Search\Controller\SearchController:
+        tags:
+            - controller.service_arguments
+
+    Ibexa\Bundle\Search\Controller\SuggestionController:
+        tags:
+            - controller.service_arguments
+
+    Ibexa\Search\Mapper\PagerSearchContentToDataMapper:
+        arguments:
+            $contentTypeService: '@ibexa.api.service.content_type'
+            $userService: '@ibexa.api.service.user'
+            $userLanguagePreferenceProvider: '@Ibexa\Core\MVC\Symfony\Locale\UserLanguagePreferenceProvider'
+            $translationHelper: '@Ibexa\Core\Helper\TranslationHelper'
+            $languageService: '@ibexa.api.service.language'
+
+    Ibexa\Search\QueryType\SearchQueryType: ~
+
+    Ibexa\Search\EventDispatcher\EventListener\ContentSuggestionSubscriber: ~
+
+    Ibexa\Search\Mapper\SearchHitToContentSuggestionMapper: ~
+
+    Ibexa\Contracts\Search\Mapper\SearchHitToContentSuggestionMapper: '@Ibexa\Search\Mapper\SearchHitToContentSuggestionMapper'
+
+    Ibexa\Search\Service\SuggestionService: ~
+
+    Ibexa\Bundle\Search\Request\ParamConverter\SuggestionQueryParamConverter:
+        arguments:
+            $defaultLimit: '%ibexa.site_access.config.default.search.suggestion.min_query_length%'
```

I'm not agree with the name `min_query_length` for this parameter. This name is misleading. The parameter is the query's limit argument, so, it's the maximum number of results. The name could be such as `max_suggestion_count`, or `suggestions_limit`.

---

It's a typo it should be: "result_limit" - fixed

---

I get it: This is not the right parameter here according to PR description.

```suggestion
            $defaultLimit: '%ibexa.site_access.config.default.search.suggestion.result_limit%'
```

---

@kisztof I understood a split second before you fixed it 😛 

What will be `ibexa.site_access.config.default.search.suggestion.min_query_length`? The minimum count of characters in the search box before starting to suggest?